### ISSUE TITLE
Enable StrictMode for spec/dummy

### ISSUE THREAD

**Issue by alexeyr-ci1** [shakacode/react_on_rails#1465]
See https://reactjs.org/docs/strict-mode.html. This should help detect more bugs in RoR.

**Comment by justin808** [comment]
@alexeyr-ci1 SGTM

**Comment by alexeyr-ci1** [comment]
@justin808 After #1463.

**Comment by ahangarha** [comment]
> ... StrictMode added some more restrictions in React 18.

In bad way? Anything particular apart from twice rendering which is deliberate?

May you please elaborate on this?

**Comment by alexeyr-ci1** [comment]
> In bad way? Anything particular apart from twice rendering which is deliberate?

No, in a good way which catches more errors, which is what we want StrictMode for.

**Comment by justin808** [comment]
Closing. StrictMode is now standard in modern React development and widely adopted in the ecosystem.

**Comment by justin808** [comment]
Reopening for investigation. Need to verify StrictMode status in spec/dummy and determine if implementation is needed.

**Comment by alexeyr-ci2** [comment]
> Closing. StrictMode is now standard in modern React development and widely adopted in the ecosystem.

That's why we should probably use it, yes.

> Need to verify StrictMode status in spec/dummy

It isn't currently used: https://github.com/search?q=repo%3Ashakacode%2Freact_on_rails%20strictmode&type=code

This issue now covers the RORP dummy too.

**Comment by justin808** [comment]
Deep-dive triage pass (2026-03-22):

- Domain: core/runtime
- Labels/Priority signals: enhancement, P3
- Current PR coverage: no open PR currently linked
- Proposed next step: add this to the new triage PR stack so it has explicit implementation ownership.

Question: Do you want this scheduled in the next implementation wave, or parked until after active release work?
